### ISSUE TITLE
[WIP] Move from cnx-recipes PDF generation scripts to bakery-cli

### ISSUE THREAD

**Issue by brittany-johnson** [openstax/cnx-recipes#2598]
# Scenario
[Bakery CLI NPM Package](https://www.npmjs.com/package/@openstax/bakery-cli)

# Acceptance Criteria (required)
- [ ] @brittany-johnson has updated or requested an updated for the install documentation for the bakery-cli npm package
All developers on ce-styles and QA have properly installed bakery-cli and ran through all steps in the pipeline to generate a PDF.
- [ ] @brittany-johnson 
- [ ] @Stackblocks 
- [ ] @wilkus27 
- [ ] @ajluk 
- [ ] @MartaSztuka 
- [ ] @meeragovindaraju 

# Steps to Reproduce (required)
* **How many times were you able to reproduce the issue?**
* **Images or Screenshots Included:** [ ] Yes  [ ] No  (_attach below_)

# Environment: (required)
* **Medium**: [ ] REX [ ] PDF  [ ] cnx.org
* **Instance**: [ ] production  [ ] qa  [ ] staging  [ ] dev  [ ] content
* **OS**: [ ] Mac OSX  [ ] Windows  [ ] iOS  [ ] Android
* **Browser**: [ ] Chrome  [ ] Firefox  [ ] Safari  [ ] Opera  [ ] IE  [ ] Edge

# Impact
* **Severity**: [ ] High [ ] Medium  [ ] Low
* **Frequency of Occurrence**: [ ] High [ ] Medium  [ ] Low

# Next Steps for User
If the card is accepted or rejected, what does user need to know now?

# Test Plan

**Comment by wilkus27** [comment]
@brittany-johnson I went through the installation process and it succeeded. I have some problems with commands. 

#### 1. I have to run all commands with sudo.

Otherwise I get such error:

![image](https://user-images.githubusercontent.com/40228854/90655313-b4a9cd00-e241-11ea-9393-9e59527a6721.png)

#### 2. It's hard to fetch the book

When I'm using WiFi from the rooter, process of fetching stuck in different moments. I have to switch to the internet from my mobile phone and then it works, but I can't do anything else during process, even write massage on Slack, because then I get error with timing out

**Comment by ajluk** [comment]
@brittany-johnson on my side the installation process also succeeded, but I had similar problem with running the fetching script as Natalia described above when was connected to my wired local network: 
![image](https://user-images.githubusercontent.com/55203598/90747277-06e5ff00-e2d1-11ea-9ac8-3953c9131fcc.png)
I've got this error from above or sometimes the fetching process just stopped in some moment. And then there was only part of the book fetched:
(baked file with uncomplete fetch)
![image](https://user-images.githubusercontent.com/55203598/90748038-f4b89080-e2d1-11ea-9d6e-53c2c528931a.png)

I have also switched to the mobile network and then it worked as expected. The rest of the scripts worked well.

**Comment by MartaSztuka** [comment]
@brittany-johnson Installation process succeeded, but I couldn't complete fetching, didn't receive any error message, it simply stuck. However other scripts went well.
![Screenshot from 2020-08-20 12-59-55](https://user-images.githubusercontent.com/47241114/90762747-d492cc80-e2e5-11ea-8106-168a652eaad2.png)

**Comment by meeragovindaraju** [comment]
Done and it worked 🎉

**Comment by brittweinstein** [comment]
Next steps: BrittanyJ will need to meet with Bey to discuss priority. We need to be using the same pipeline, but there are many other high priority items in progress right now.

**Comment by therealmarv** [comment]
Question @wilkus27 @ajluk @MartaSztuka

When you ran your tests in August were you all connected to the same internal network?
Are your errors reproducible e.g. from your home internet?

This part makes me wonder why if this is maybe an internal network problem `When I'm using WiFi from the rooter, process of fetching stuck in different moments. I have to switch to the internet from my mobile phone and then it works`.

**Comment by ajluk** [comment]
@therealmarv No, we were not connected to the same internal network. Our whole team is working remotely since the pandemic started to spread (it was in march in Poland). We encountered the errors while been connected to our home internet. For all three of us it is the same internet provider though (UPC). It's a coincidence but maybe it is relevant. Here is my fiber connection speed details if it can help: https://www.speedtest.net/result/9951211475

**Comment by rnathuji** [comment]
@ajluk - Out of curiosity: What throttle limit value do you typically use with the `fetch-book` script / `neb` to avoid this issue with the old pipeline? Is this typically something you specify regularly, or only if it fails otherwise? 

I'm also going to try and tune these values a bit to see if we can reproduce that way in our test environments.

**Comment by ajluk** [comment]
@rnathuji I mainly use `--request-limit 2` ( whole script: `docker-compose run --rm fetch-book --request-limit 2 --with-resources`). I need to use this flag every time I run the fetch script otherwise it got stuck at some point. 
`--request-limit 1` doesn't work for me either. So I guees it needs to be 2 or bigger.

**Comment by rnathuji** [comment]
@ajluk - Thanks for the information! A quick additional request: could you measure the latency from your home internet to `archive.cnx.org` using `ping archive.cnx.org -c 5` (you can just provide the output from that).

**Comment by ajluk** [comment]
@rnathuji here is the outcome:
ajluk@msib450:~$ ping archive.cnx.org -c 5
PING prod00.cnx.org (128.42.169.27) 56(84) bytes of data.
64 bytes from cnx.org (128.42.169.27): icmp_seq=1 ttl=47 time=158 ms
64 bytes from cnx.org (128.42.169.27): icmp_seq=2 ttl=47 time=159 ms
64 bytes from cnx.org (128.42.169.27): icmp_seq=3 ttl=47 time=162 ms
64 bytes from cnx.org (128.42.169.27): icmp_seq=4 ttl=47 time=156 ms
64 bytes from cnx.org (128.42.169.27): icmp_seq=5 ttl=47 time=160 ms

--- prod00.cnx.org ping statistics ---
5 packets transmitted, 5 received, 0% packet loss, time 4003ms
rtt min/avg/max/mdev = 156.062/159.043/162.171/2.045 ms
ajluk@msib450:~$ `
`

**Comment by rnathuji** [comment]
@ajluk , @wilkus27 , @MartaSztuka - We've released a version `0.0.7` of the [bakery-cli](https://www.npmjs.com/package/@openstax/bakery-cli) which adds the `--request-limit` option to the `fetch` command. So, for example, you can run `bakery run fetch cnx.org col30939 latest -d data --request-limit 2` in the same manner as you used this parameter in the earlier scripts. 

Can you update your version of the CLI and give this a try to see if it helps address your issues pulling content with the bakery CLI?

**Comment by ajluk** [comment]
@rnathuji Thank you for letting us know! That's good news! I will check the new version out on monday and give you a feedback how it goes.

**Comment by ajluk** [comment]
@rnathuji The fetching process has succeeded with flag `--request-limit 2`! Thank you! I needed to change the sequence though in the command you provided above to `bakery run fetch katalyst01.cnx.org col25734 latest --request-limit 2 -d ./data/intro-business`

**Comment by MartaSztuka** [comment]
@rnathuji I used the same scripts as Anna, worked well, thanks!

**Comment by rnathuji** [comment]
@wilkus27 - Have you had a chance to test the CLI `fetch` command using the `--request limit` flag? I just wanted to confirm things are working for you as well before closing out [this issue](https://github.com/openstax/cnx/issues/1203).

**Comment by wilkus27** [comment]
@rnathuji sorry for the delay. I had some problems with scripts but they were resolved on the Slack. I'm leaving here details just for information and to keep everything in one place.

After running fetch command with limit request I got such error:

![image](https://user-images.githubusercontent.com/40228854/101361097-d84f5c80-389e-11eb-8567-307d05f26e50.png)

Command `bakery stop -d` helped.

I was able to run all other scripts. Problems appeared with mathify script:

![image](https://user-images.githubusercontent.com/40228854/101361307-1c426180-389f-11eb-9a77-76d455a1b1aa.png)

I had to run `bakery stop -d` again.

Everything is working now for me.

**Comment by lwtchu** [comment]
:eyes: i think we can close this. if this is not good you can reopen or park with us or something.
